### PR TITLE
Bugfixes

### PR DESCRIPTION
**Pull request by grechaw**
This PR fixes three issues.

Function signatures for instances have changed -- one commit has fix and tests for #240 

TDE template update to handle primary keys that are not strings, #247 

And the search options issue discovered by @bsrikan in which search options with just one range index generated an invalid template.

### CODE REVIEW

**Comment by grechaw** [comment]
Note version translator test has a failure due to a server bug.

File: entity-services/src/main/xdmp/entity-services/entity-services-impl.xqy [hunk 1]
```diff
@@ -1008,7 +1010,7 @@ declare function esi:extraction-template-generate(
                                 <tde:column>
                                     { comment { "This column joins to primary key of",
                                                 $ref-type-name } }
-                                    <tde:name>{ $ref-primary-key }</tde:name>
+                                    <tde:name>{ $property-name || "_" || $ref-primary-key }</tde:name>
```
Comment (bsrikan) [inline entity-services/src/main/xdmp/entity-services/entity-services-impl.xqy]: Just FYI... I have in my tdex removed $property-name_ for the column name and it works OK.

**Comment by bsrikan** [comment]
Testing the PR. Updated functional tests to handle tde primary key datatype issue. Verifying other 2 bug fixes now.

**Comment by grechaw** [comment]
Regarding latest, had to push work in progress before tests were done.  It's for bug 214, where the existing strategy creates duplicates sometimes

**Comment by grechaw** [comment]
Note I've done a `push -f` on this branch.  The four bugs I set out to do this week are all in this PR now.

**Comment by bsrikan** [comment]
This PR is good to go @grechaw

**Comment by grechaw** [comment]
Thanks @bsrikan , several issues coming your way.

**Comment by bsrikan** [comment]
No problem. Thats why I took the time to update functional tests per fixes in this PR.

**Pull request merged**